migrations/update_legacy_filtered_positives.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import logging.config
+from datetime import datetime
+from crawler.helpers.general_helpers import get_config
+from migrations.helpers.update_filtered_positives_helper import (
+    update_filtered_positive_fields,
+    update_mlwh_filtered_positive_fields,
+    update_mongo_filtered_positive_fields,
+)
+from migrations.helpers.update_legacy_filtered_positives_helper import (
+    legacy_mongo_samples,
+    get_cherrypicked_samples_by_date,
+    v0_version_set,
+    split_mongo_samples_by_version,
+)
+from crawler.constants import (
+    V0_V1_CUTOFF_TIMESTAMP,
+    V1_V2_CUTOFF_TIMESTAMP,
+)
+from crawler.filtered_positive_identifier import (
+    FILTERED_POSITIVE_VERSION_0,
+    FILTERED_POSITIVE_VERSION_1,
+    FILTERED_POSITIVE_VERSION_2,
+)
+from migrations.helpers.dart_samples_update_helper import extract_required_cp_info
+
+logger = logging.getLogger(__name__)
+
+# Migration steps:
+# 1. Get all legacy samples (those created in Mongo prior to Crawler
+#    setting the filtered positive fields) from Mongo
+# 2. Query to find which of these samples belong to v0, v1, v2 based on
+#    when they were created in the 'sample' table of MLWH
+# 3. Update the filtered positive fields of the samples using the correct
+#    version rules
+# 4. Update Mongo and MLWH with these filtered positive fields
+
+
+def run(settings_module: str = "") -> None:
+    """Migrate the existing samples to have the filtered positive values.
+
+    Arguments:
+        config {ModuleType} -- application config specifying database details
+    """
+    config, settings_module = get_config(settings_module)
+    logging.config.dictConfig(config.LOGGING)  # type: ignore
+
+    logger.info("-" * 80)
+    logger.info("STARTING FILTERED POSITIVES LEGACY UPDATE")
+    logger.info(f"Time start: {datetime.now()}")
+
+    mongo_versions_updated = {
+        FILTERED_POSITIVE_VERSION_0: False,
+        FILTERED_POSITIVE_VERSION_1: False,
+        FILTERED_POSITIVE_VERSION_2: False,
+    }
+
+    mlwh_versions_updated = {
+        FILTERED_POSITIVE_VERSION_0: False,
+        FILTERED_POSITIVE_VERSION_1: False,
+        FILTERED_POSITIVE_VERSION_2: False,
+    }
+
+    try:
+        if v0_version_set(config):
+            question = "v0 version has been set on some samples. This migration has likely been \
+                        run before - do you still wish to proceed? (yes/no):"
+            continue_migration = get_input(question)
+
+            if continue_migration == "yes":
+                pass
+            elif continue_migration == "no":
+                logger.info("Now exiting migration")
+                raise Exception()
+            else:
+                logger.info("Invalid input, please enter 'yes' or 'no'. Now exiting migration")
+                raise Exception()
```

Comment:
I don't particularly like that exceptions are raised here. Could you instead set a local bool variable that you check before line 79 that just skips lines 79-140, resulting in hitting the `finally` block without having to raise an exception?

Reply:
Yep, makes sense, I've changed to use a bool instead